qa/workunits/rados/test_dedup_tool.sh

```diff
@@ -242,9 +242,58 @@ function test_dedup_chunk_repair()
   $RADOS_TOOL -p $POOL rm bar
 }
 
+function test_dedup_object()
+{
+
+  CHUNK_POOL=dedup_chunk_pool
+  run_expect_succ "$CEPH_TOOL" osd pool create "$CHUNK_POOL" 8
+
+  echo "There hiHI" > foo
+
+  $RADOS_TOOL -p $POOL put foo ./foo
+
+  sleep 2
+
+  rados ls -p $CHUNK_POOL
+
+  RESULT=$($DEDUP_TOOL --pool $POOL --op chunk-dedup --object foo --chunk-pool $CHUNK_POOL --source-off 0 --source-length 10 --fingerprint-algorithm sha1 )
+
+  POOL_ID=$($CEPH_TOOL osd pool ls detail | grep $POOL |  awk '{print$2}')
+  CHUNK_OID=$(echo -n "There hiHI" | sha1sum | awk '{print $1}')
+
+  RESULT=$($DEDUP_TOOL --op dump-chunk-refs --chunk-pool $CHUNK_POOL --object $CHUNK_OID | grep foo)
+
+  if [ -z "$RESULT" ] ; then
+    $CEPH_TOOL osd pool delete $POOL $POOL --yes-i-really-really-mean-it
+    $CEPH_TOOL osd pool delete $CHUNK_POOL $CHUNK_POOL --yes-i-really-really-mean-it
+    die "Scrub failed expecting bar is removed"
+  fi
```

Comment:
Shouldn't we try to read the object?

Reply:
I added codes to read and verify the chunk object.